csfieldguide/static/interactives/2d-shape-manipulations/js/2d-shape-manipulations.js

```diff
@@ -0,0 +1,894 @@
+    });
+    $('#matrix-first-translate-row-1-col-0').on('blur', function() {
+        matrixTab('matrix-first-translate-row-1-col-0');
+    });
+    $('#matrix-second-scale-row-0-col-0').on('blur', function() {
+        matrixTab('matrix-second-scale-row-0-col-0');
+    });
+    $('#matrix-second-scale-row-0-col-1').on('blur', function() {
+        matrixTab('matrix-second-scale-row-0-col-1');
+    });
+    $('#matrix-second-scale-row-1-col-0').on('blur', function() {
+        matrixTab('matrix-second-scale-row-1-col-0');
+    });
+    $('#matrix-second-scale-row-1-col-1').on('blur', function() {
+        matrixTab('matrix-second-scale-row-1-col-1');
+    });
+    $('#matrix-second-translate-row-0-col-0').on('blur', function() {
+        matrixTab('matrix-second-translate-row-0-col-0');
+    });
+    $('#matrix-second-translate-row-1-col-0').on('blur', function() {
+        matrixTab('matrix-second-translate-row-1-col-0');
+    });
+}
+
+
+/* Registers the handler function for when the user presses return */
+// Solution from https://stackoverflow.com/questions/979662/how-to-detect-pressing-enter-on-keyboard-using-jquery
+function registerEnterHandler() {
+    $(document).on('keypress', function(key) {
+        if(key.which == 13) {
+            // Enter was pressed
+            runEnterHandler();
+        }
+    });
+}
+
+
+/* If the currently active element is an input box, updates the shape and activates the next input box */
+function runEnterHandler() {
+    var activeElement = document.activeElement;
+    var id = activeElement.id;
+    var nextElementId = '';
+    var elementVariable1;
+    var elementVariable2;
+    if (id.includes('-input-')) {
+        // Element is a coordinate input
+        // pN-input-V Where N is the node and V is x or y
+        elementVariable1 = parseInt(id.substring(1, 2));    // N
+        elementVariable2 = id.substring(9);                 // V
+
+        var nextNode = elementVariable2 == 'x' ? elementVariable1 + 0 : elementVariable1 + 1;
+        if (nextNode > 6) {
+            nextNode = 0;
+        }
+        var nextAxis = elementVariable2 == 'x' ? 'y' : 'x';
+        nextElementId = '#p' + nextNode + '-input-' + nextAxis;
+
+        coordTab('c' + elementVariable1, elementVariable2);
+        $(nextElementId).focus();
+
+    } else if (id.includes('-row-')) {
+        // Element is a matrix input
+        // matrix-NAME-TYPE-row-X-col-Y Where NAME, TYPE specify which matrix it is, and (X,Y) is the box's location within the matrix
+        if (id.includes('first')) {
+            elementVariable1 = ['first', (id.includes('scale') ? 'scale' : 'translate')];
+        } else {
+            elementVariable1 = ['second', (id.includes('scale') ? 'scale' : 'translate')]
+        }
+
+        var indexRow = id.indexOf('row-');
+        var indexCol = id.indexOf('col-');
+        elementVariable2 = [id.substring(indexRow + 4, indexRow + 5), id.substring(indexCol + 4, indexCol + 5)]
```

Comment:
We should try to avoid 'magic' numbers where possible. Maybe add a comment explaining that you are stripping 'row-' or 'col-' or something along those lines etc